ignite/chainconfig/v0/configyaml.go

```diff
@@ -0,0 +1,105 @@
+package v0
+
+import "github.com/ignite-hq/cli/ignite/chainconfig/common"
+
+// ConfigYaml is the user given configuration to do additional setup
+// during serve.
+type ConfigYaml struct {
+	Accounts              []common.Account       `yaml:"accounts"`
+	Validator             common.Validator       `yaml:"validator"`
+	Faucet                common.Faucet          `yaml:"faucet"`
+	Client                common.Client          `yaml:"client"`
+	Build                 common.Build           `yaml:"build"`
+	Init                  common.Init            `yaml:"init"`
+	Genesis               map[string]interface{} `yaml:"genesis"`
+	Host                  common.Host            `yaml:"host"`
+	common.BaseConfigYaml `yaml:",inline"`
+}
+
+// AccountByName finds account by name.
+func (c ConfigYaml) AccountByName(name string) (acc common.Account, found bool) {
+	for _, acc := range c.Accounts {
+		if acc.Name == name {
+			return acc, true
+		}
+	}
+	return common.Account{}, false
+}
+
+// GetVersion returns the version of the config.yaml file.
+func (c ConfigYaml) GetVersion() string {
+	return c.Version
+}
+
+// GetFaucet returns the Faucet.
+func (c ConfigYaml) GetFaucet() common.Faucet {
+	return c.Faucet
+}
+
+// GetBuild returns the Build.
+func (c ConfigYaml) GetBuild() common.Build {
+	return c.Build
+}
+
+// GetHost returns the Host.
+func (c ConfigYaml) GetHost() common.Host {
+	return c.Host
+}
+
+// GetGenesis returns the Genesis.
+func (c ConfigYaml) GetGenesis() map[string]interface{} {
+	return c.Genesis
+}
+
+// GetInit returns the Init.
+func (c ConfigYaml) GetInit() common.Init {
+	return c.Init
+}
+
+// GetClient returns the Client.
+func (c ConfigYaml) GetClient() common.Client {
+	return c.Client
+}
+
+// ListAccounts returns the list of all the accounts.
+func (c ConfigYaml) ListAccounts() []common.Account {
+	return c.Accounts
+}
+
+// ListValidators returns the list of all the validators.
+func (c ConfigYaml) ListValidators() []common.Validator {
+	return []common.Validator{c.Validator}
+}
+
+// Clone returns an identical copy of the instance
+func (c *ConfigYaml) Clone() common.Config {
+	copy := *c
```

Comment:
This is probably not an effective copy.

Reply:
It actually is effective. I can add tests.

Reply:
It's *somewhat* effective. Depending on what you're trying to copy. If your struct only has concrete simple types (int, string, etc) then it works great.

But if your struct contains references (pointers), then this approach doesn't apply any "deep" copy semantics. 

This is OK for the v0 config, but the v1 config contains a slice of `Validators`, and since Go treats any slice as a pointer, this copy approach falls short, and the copied version will be referencing the same slice as the original.